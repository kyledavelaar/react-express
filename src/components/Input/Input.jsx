// @flow

import * as React from 'react';

import s from './Input.scss';

type Props = {
  label: string|number,
  type: string|number,
  value: string|number,
  name: string|number,
  placeholder: string|number,
  checked: boolean,
  onChange: (e: SyntheticEvent<any>) => void,
}

export const Input = (props: Props) => {
  const { label, type, value, name, placeholder, checked, onChange } = props;
  const style = getStyle(type);

  return (
    <label className={s.label}>
      { type === 'text' && label }
      <input
        className={style.join(' ')} 
        type={type}
        value={value}
        name={name}
        placeholder={placeholder}
        checked={checked}
        onChange={onChange}
      />
      { type !== 'text' && label }
    </label>
  )
}

function getStyle(type: string|number) {
  let style = [s.input]
  if (type !== 'text') {
    style.push(`${s.checkbox}`)
  }
  return style;
}


export default Input;