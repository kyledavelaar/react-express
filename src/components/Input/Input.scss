@import '../../../variables.scss';


.label {
  margin: 10px 0;
  color: $graphite;
}

.input {
  font-size: 20px;
  padding: 5px 15px;
  border-radius: 4px;
  border: 1px solid lightgrey;
  margin: 10px 0;
  width: 400px;
  color: $graphite;
  display: block;
}

.checkbox {
  display: inline;
  margin-right: 5px;
  width: inherit;
}