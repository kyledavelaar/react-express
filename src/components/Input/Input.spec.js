import React from 'react';
import { mount, shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import { Input } from './Input';

Enzyme.configure({ adapter: new Adapter() })


describe('Input renders',() => {

  it('Input component created', () => {
    const rendered = renderer.create(
      <Input 
      type={'text'}
      value={'my text'}
      name={'username'}
      placeholder={'input your username'}
      checked={false}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  })

})
