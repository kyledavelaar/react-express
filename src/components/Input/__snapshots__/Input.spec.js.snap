// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders Input component created 1`] = `
<label
  className={undefined}
>
  <input
    checked={false}
    className=""
    name="username"
    onChange={undefined}
    placeholder="input your username"
    type="text"
    value="my text"
  />
</label>
`;
